cpp/examples/arrow/dataset-parquet-scan-example.cc

```diff
@@ -65,7 +65,7 @@ struct Configuration {
   std::shared_ptr<ds::Expression> filter = ("total_amount"_ > 1000.0f).Copy();
 } conf;
 
-std::shared_ptr<ds::Dataset> GetDatasetFromPath(fs::FileSystem* fs,
+std::shared_ptr<ds::Dataset> GetDatasetFromPath(fs::FileSystemPtr fs,
```

Comment:
Hmm, I'm not a huge fan of this pattern was this already existing in the code base or was part of this change to explicitly use it?  It seems like it is something that should have possibly been discussed before hand? (aplogies if I missed it).

Reply:
Aliases are also discouraged by the style guide: https://google.github.io/styleguide/cppguide.html#Aliases

Reply:
Per discussions on the mailing list I also am not a fan of having aliases in the public API